### ISSUE TITLE
Make panel titles horizontal

### ISSUE THREAD

**Issue by ebugden** [theia-ide/theia-trace-extension#154]
Horizontal text is the fastest and most comfortable to read (compared to vertical, diagonal, etc.) 

Position the ‘X’ for deleting the chart at the far right (keeps deletion intuitive by following the standard for horizontal windows in VS Code).

![horizontal-panel-title](https://user-images.githubusercontent.com/28311615/95917093-b2627b80-0d77-11eb-969a-49bcd069c21a.PNG)

Other benefits:
* Horizontal titles can structure the layout by doubling as visual dividers between each panel
* Horizontal orientation leaves room for longer titles

Quick test you can try on yourself to show that horizontal text is significantly faster to read ([Article](https://jov.arvojournals.org/article.aspx?articleid=2121153)). 

![horizontal-vertical-text-reading2](https://user-images.githubusercontent.com/28311615/95917612-872c5c00-0d78-11eb-93b2-c25f2d2f132a.PNG)

**Comment by mirsky-work** [comment]
+1 for the suggestion.
Is there any specific reason why the Panel Title is aligned to the left and not centered (I don't disagree, just curious)?

**Comment by ebugden** [comment]
Good question. Honestly left alignment was just my first reflex, but in retrospect it seems like a safe default. 

An intentional decision about alignment could be made later on. I think the main decision factors would be:
* **Title length:** How many details about the data source are included in the panel title? How much does length vary? 
* **Panel layout & complexity:** Does the title feel too far from the graph (loose visual association)? How many things are distracting the user from seeing the title?

# Left aligned titles
**Consistent title start position** regardless of length:
* Fast to scan vertically
* Organized look even with collapsed panels 

![left-aligned-titles-collapsed](https://user-images.githubusercontent.com/28311615/96023898-15f4b380-0e21-11eb-8552-615a645ded00.PNG)

# Centered titles
**Inconsistent title start position:** 
* Eye travels more while scanning
* Can look more messy

![centered-titles-collapsed](https://user-images.githubusercontent.com/28311615/96023895-15f4b380-0e21-11eb-85cc-81b8d768287a.PNG)

# Counter example: Grafana
I think centered titles could be considered if we move away from a simple vertically stacked panel layout with collapsible panels towards a more complex layout. Grafana's centered titles work well in their case because they have a:
* **More complex layout with variable panel sizes:** The centered titles catch the eye quickly (_because_ they feel disconnected from from rest of the nicely aligned layout) and give you a sense of the overall dashboard layout as well as the panel sizes.

This is supported by: 
* Minimal panel design (low visual noise)
* Short high-level titles
* Panels are not collapsible

![grafana-dashboard-example](https://user-images.githubusercontent.com/28311615/96025089-a8498700-0e22-11eb-872d-cfdec2c8b19e.PNG)

**Here's left aligned Grafana for comparison.** I actually feel a bit more lost here because the colourful data draws my eye first and it takes me bit longer to find the panel title. 

![grafana-left-aligned-titles](https://user-images.githubusercontent.com/28311615/96030658-44c35780-0e2a-11eb-8813-fd30c8abf926.png)

**Comment by bhufmann** [comment]
I agree having the title on top of the graphs seems to provide a better user experience. 

The reason why the initial implementation of the tool had the title on the left and vertically, was to save vertical screen space. Trace Compass has received the feedback that there is too much vertical space lost due views toolbar and title. Users often want to correlate different graphs by putting them under each other on the same time axis and have them close to each other.

Maybe we'd like to have an configuration option for the user to hide the title bar if wanted.

**Comment by ebugden** [comment]
## Spending vertical space on titles: More difficult to compare many graphs?
> bhufmann: **Users often want to correlate different graphs** by putting them under each other on the same time axis and have them close to each other.

My expectation is that we can still deliver on that desire for easily comparable and information rich layouts even with some of that space spent on horizontal titles. For example, I think being able to easily add, reorder, and collapse many data-focused panels with a minimal style will go a long way to helping users feel like they can "lay everything they want to compare out on the table".

I would also argue that a title (or some sort of data label) is one thing you always want to have visible (to avoid forgetting or mixing up the data you're looking at). 

## Allow users to hide panel titles?
> bhufmann: Maybe we'd like to have an **configuration option for the user to hide the title bar** if wanted.

Before we consider giving users the option to hide titles, I'd like to wait until closer to MVP and ask users how they feel about the information density of the new trace viewer layout.

My hope is that the frustration with the space spent on titles can be resolved with good UX choices and visual prioritization of information. Especially since, at first glance, it seems like Trace Compass and Grafana are spending similar amounts of vertical space on titles.

**Comment by bhufmann** [comment]
> ## Spending vertical space on titles: More difficult to compare many graphs?
> 
> > bhufmann: **Users often want to correlate different graphs** by putting them under each other on the same time axis and have them close to each other.
> 
> My expectation is that we can still deliver on that desire for easily comparable and information rich layouts even with some of that space spent on horizontal titles. For example, I think being able to easily add, reorder, and collapse many data-focused panels with a minimal style will go a long way to helping users feel like they can "lay everything they want to compare out on the table".
> 
I just provided some background and what feedback Trace Compass got from users about its vertical screen usage. I agree to start with the title panel on the top (and centered). 

> I would also argue that a title (or some sort of data label) is one thing you always want to have visible (to avoid forgetting or mixing up the data you're looking at).
> ## Allow users to hide panel titles?
> 
> > bhufmann: Maybe we'd like to have an **configuration option for the user to hide the title bar** if wanted.
> 
> Before we consider giving users the option to hide titles, I'd like to wait until closer to MVP and ask users how they feel about the information density of the new trace viewer layout.
> 
> My hope is that the frustration with the space spent on titles can be resolved with good UX choices and visual prioritization of information. Especially since, at first glance, it seems like Trace Compass and Grafana are spending similar amounts of vertical space on titles.

I agree. Let's get some feedback. Maybe we don't need the hiding after at all. It was just thinking ahead.
